ballerina/variable_validator.bal

```diff
@@ -128,13 +128,29 @@ class VariableValidator {
         if argument.getKind() == parser:T_IDENTIFIER {
             argument.setValue(value);
         } else if getTypeNameFromValue(value) == getTypeName(argument) {
-            argument.setValue(value);
+            if getTypeNameFromValue(value) == STRING {
+                if self.hasInvalidCharaters(<string>value) {
```

Comment:
Shall we use `value:ensureType(v, string)` to safely cast variables ?

Reply:
Since we have checked the value at line 131, Could it be a problem use direct casting here?